Improve local test running experience

These are some problems I meet at my place when running tests locally. I intend to solve them for myself and hopefully other people who meet the same problems.

1. `server::service::diagnostics::sys::tests::test_load_info` cannot pass on new Linux kernels (4.19+). The OS may provide more IO load info than the assertion expects:
```
thread 'server::service::diagnostics::sys::tests::test_load_info' panicked at 'assertion failed: `(left == right)`
  left: `["read_io/s", "read_merges/s", "read_sectors/s", "read_ticks/s", "write_io/s", "write_merges/s", "write_sectors/s", "write_ticks/s", "in_flight/s", "io_ticks/s", "time_in_queue/s", "discard_io/s", "discard_merged/s", "discard_sectors/s", "discard_ticks/s"]`,
 right: `["read_io/s", "read_merges/s", "read_sectors/s", "read_ticks/s", "write_io/s", "write_merges/s", "write_sectors/s", "write_ticks/s", "in_flight/s", "io_ticks/s", "time_in_queue/s"]`', src/server/service/diagnostics/sys.rs:633:13
```

2. We need to specify the `testexport` feature explicitly in order to compile tests of TiKV, otherwise `Api::CLIENT_TAG` cannot be found:
```
error[E0599]: no associated item named `CLIENT_TAG` found for type parameter `Api` in the current scope
    --> src/storage/mod.rs:4476:31
     |
4476 |             api_version: Api::CLIENT_TAG,
     |                               ^^^^^^^^^^ associated item not found in `Api`
```
Because of this, VSCode users may fail to use the inline "Run Test" feature.

3. `plugin_registry` tests assumes the build directory is `./target/{debug, release}/deps`. It's possible that the working directory is not the cargo root, or the user changes the default target directory path.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -165,6 +165,7 @@ seahash = "4.1.0"
 pprof = { version = "^0.6", default-features = false, features = ["flamegraph", "protobuf-codec"] }
 
 [dev-dependencies]
+api_version = { path = "components/api_version", features = ["testexport"] }
 example_plugin = { path = "components/test_coprocessor_plugin/example_plugin" } # should be a binary dependency
 panic_hook = { path = "components/panic_hook" }
 reqwest = { version = "0.11", features = ["blocking"] }

--- a/src/coprocessor_v2/plugin_registry.rs
+++ b/src/coprocessor_v2/plugin_registry.rs
@@ -466,12 +466,9 @@ mod tests {
     use coprocessor_plugin_api::util::pkgname_to_libname;
 
     fn initialize_library() -> PathBuf {
-        Path::new(if cfg!(debug_assertions) {
-            "target/debug/deps"
-        } else {
-            "target/release/deps"
-        })
-        .join(pkgname_to_libname("example-plugin"))
+        let mut path = std::env::current_exe().unwrap();
+        path.set_file_name(pkgname_to_libname("example-plugin"));
+        path
     }
 
     #[test]

--- a/src/server/service/diagnostics/sys.rs
+++ b/src/server/service/diagnostics/sys.rs
@@ -630,22 +630,19 @@ mod tests {
                 .iter()
                 .map(|x| x.get_key())
                 .collect::<Vec<&str>>();
-            assert_eq!(
-                keys,
-                vec![
-                    "read_io/s",
-                    "read_merges/s",
-                    "read_sectors/s",
-                    "read_ticks/s",
-                    "write_io/s",
-                    "write_merges/s",
-                    "write_sectors/s",
-                    "write_ticks/s",
-                    "in_flight/s",
-                    "io_ticks/s",
-                    "time_in_queue/s",
-                ]
-            );
+            assert!(keys.starts_with(&[
+                "read_io/s",
+                "read_merges/s",
+                "read_sectors/s",
+                "read_ticks/s",
+                "write_io/s",
+                "write_merges/s",
+                "write_sectors/s",
+                "write_ticks/s",
+                "in_flight/s",
+                "io_ticks/s",
+                "time_in_queue/s",
+            ]));
         }
     }
 

--- a/src/server/service/diagnostics/sys.rs
+++ b/src/server/service/diagnostics/sys.rs
@@ -630,19 +630,23 @@ mod tests {
                 .iter()
                 .map(|x| x.get_key())
                 .collect::<Vec<&str>>();
-            assert!(keys.starts_with(&[
-                "read_io/s",
-                "read_merges/s",
-                "read_sectors/s",
-                "read_ticks/s",
-                "write_io/s",
-                "write_merges/s",
-                "write_sectors/s",
-                "write_ticks/s",
-                "in_flight/s",
-                "io_ticks/s",
-                "time_in_queue/s",
-            ]));
+            assert!(
+                keys.starts_with(&[
+                    "read_io/s",
+                    "read_merges/s",
+                    "read_sectors/s",
+                    "read_ticks/s",
+                    "write_io/s",
+                    "write_merges/s",
+                    "write_sectors/s",
+                    "write_ticks/s",
+                    "in_flight/s",
+                    "io_ticks/s",
+                    "time_in_queue/s",
+                ]),
+                "actual: {:?}",
+                keys
+            );
         }
     }
 

